src/LanguageServer/Protocol/Features/Diagnostics/DocumentAnalysisExecutor_Helpers.cs

```diff
@@ -127,84 +128,6 @@ static string GetLanguageSpecificId(string? language, string noLanguageId, strin
                 language: language);
         }
 
-        public static async Task<CompilationWithAnalyzersPair?> CreateCompilationWithAnalyzersAsync(
```

Comment:
moved to local function in single place it is needed.

Reply:
i also lifted the 'compilation == null' check below out of it.  but it is otherwise unchanged.